src/shared/database/index.ts

```diff
@@ -482,6 +460,46 @@ class Pindexer {
       .where('transaction_id', '=', Buffer.from(hexToUint8Array(txHash)))
       .executeTakeFirst();
   }
+
+  async queryLeaderboard(limit: number, interval: string, baseHex?: string, quoteHex?: string) {
```

Comment:
thoughts: in my initial PR version, I tried to split this large database file and bring the `queryLeaderboard` function to `pages/leaderboard/api` where it logically belongs according to the FSD. I don't mind having it in this file and not pushing for refactor here, but with time it would be nice to split it

Reply:
Noted.